src/libs/ajax/User.test.ts

```diff
@@ -1,38 +1,441 @@
+  programLocationCity: completeUserProfile.programLocationCity,
+  programLocationState: completeUserProfile.programLocationState,
+  programLocationCountry: completeUserProfile.programLocationCountry,
+  researchArea: completeUserProfile.researchArea,
+};
+
+const NA = 'N/A';
+
+// create user profile
+describe('A create user profile request', () => {
+  describe('when minimally filled out by the user', () => {
+    // when a user has filled out the required fields of a profile
+    // and has not entered values for the optional fields
+    // those optional fields should be changed to 'N/A' in the request
+    it.each(['title', 'institute', 'programLocationCity', 'programLocationState', 'programLocationCountry'])(
+      'sets undefined property %s to "N/A"',
+      async (field) => {
+        // Arrange, Act
+        const apiBody: OrchestrationUpsertTerraUserProfileRequest = generateAPIBodyForCreateUserProfile(
+          minimalCreateUserProfileRequest
+        );
+
+        // Assert
+        expect(apiBody[field]).toBe(NA);
+      }
+    );
+    describe('should not modify property', () => {
+      // these required fields that the user filled out should not be changed
+      it('interestInTerra', async () => {
+        // Arrange, Act
+        const apiBody: OrchestrationUpsertTerraUserProfileRequest = generateAPIBodyForCreateUserProfile(
+          minimalCreateUserProfileRequest
+        );
+
+        // Assert
+        expect(apiBody.interestInTerra).toBe(minimalCreateUserProfileRequest.interestInTerra);
+      });
+      it('department', async () => {
+        // Arrange, Act
+        const apiBody: OrchestrationUpsertTerraUserProfileRequest = generateAPIBodyForCreateUserProfile(
+          minimalCreateUserProfileRequest
+        );
+
+        // Assert
+        expect(apiBody.department).toBe(minimalCreateUserProfileRequest.department);
+      });
+    });
+    describe('should have undefined properties', () => {
+      // because the user is registering for the first time research area should be 'undefined'
+      // these are not required to be initialized as 'N/A'
+      it('researchArea', async () => {
+        // Arrange, Act
+        const apiBody: OrchestrationUpsertTerraUserProfileRequest = generateAPIBodyForCreateUserProfile(
+          minimalCreateUserProfileRequest
+        );
+
+        // Assert
+        expect(apiBody.researchArea).toBe(undefined);
+      });
+    });
+    describe('should not modify or have undefined property', () => {
+      // these properties should not be modified and should not have a value of 'N/A'
+      it('firstName', async () => {
+        // Arrange, Act
+        const apiBody: OrchestrationUpsertTerraUserProfileRequest = generateAPIBodyForCreateUserProfile(
+          minimalCreateUserProfileRequest
+        );
+
+        // Assert
+        expect(apiBody.firstName).toBe(completeUserProfile.firstName);
+        expect(apiBody.firstName).not.toBe(undefined);
```

Comment:
Nit: [toBeDefined](https://jestjs.io/docs/expect#tobedefined) would read a little better than `not.toBe(undefined)` or `not.toBeUndefined()`.

```suggestion
        expect(apiBody.firstName).toBeDefined();
```